_posts/2023-07-10-custom-http-signaling-notification.adoc

```diff
@@ -0,0 +1,249 @@
+---
+layout: post
+title: "Debezium signaling and notifications - Part 2: Customisation"
+date: 2023-07-10
+tags: [ debezium, features, notifications, signaling, custom, channels ]
+featured: true
+author: anmohant
+---
+
+Welcome to this series of articles dedicated to signaling and notifications in Debezium! This post serves as the second installment in the series, where we will discuss how to customize the signal and notification channels in Debezium.
+
+Debezium 2.3 introduced new improvements in signaling and notification capabilities. You can set up new signals and notification channels in addition to the pre-defined signals and notification channels offered by Debezium. This feature enables users to customize the system to suit their unique needs and combine it with their existing infrastructure or third-party solutions.
+It enables effective monitoring and a proactive response to data changes by precisely capturing and communicating signal events and triggering notifications through preferred channels.
+
+The first article in this series, [Signaling and Notifications in Debezium](https://debezium.io/blog/2023/06/27/Debezium-signaling-and-notifications/), provides an overview of the signaling and notification features in Debezium. It also discusses the available channels & their use cases for various scenarios.
+
++++<!-- more -->+++
+
+== Custom Signal & Notification Channel
+
+In Debezium, it is possible to customize the signal and notification channels to fit specific requirements. For example, we can achieve a customization by creating an `HTTP` channel for both signals and notifications. This `HTTP` channel receives signals from an http endpoint, and notifications can be sent back to the endpoint upon the delivery of the signal.
+
+Let's explore an example that demonstrates how to create and utilize the `HTTP` signal and notification channels using the Debezium Postgres connector, a Mock Server, and [Postbin](https://www.toptal.com/developers/postbin/) which is used as the HTTP endpoint to receive the notifications.
+
+=== Setting up the `HTTP` Signal Channel:
+
+- Configure the Debezium Postgres connector to emit signals whenever a relevant database change occurs.
+- Set up a Mock Server as a receiver for these signals. The Mock Server acts as an intermediary, capturing the signals and forwarding them to the desired http endpoint.
+- Configure the Mock Server to forward the signals to an http endpoint using the appropriate HTTP method (e.g., POST).
+- Customize the `HTTP` channel settings to define the HTTP endpoint URL, authentication, headers, and any additional parameters as needed.
+
+=== Setting up the `HTTP` Notification Channel:
+
+- Once the signal is received by the Mock Server, it can trigger the posting of a notification to an http endpoint using Postbin or any other suitable http endpoint.
```

Comment:
Also here, in a more generic way the `HttpNotificationChannel` simply sends Debezium `Notification` to an HTTP endpoint, that for this example is `Postbin`.